### PR TITLE
Fix unicode in DDRec and add python import test

### LINKED ISSUE #563: Python compatibility check with ROOT
Add python compatibility check between the version that was used to build ROOT and the version CMake has found for the build of DD4hep. Requires ROOT 6.20 and this [ticket](https://sft.its.cern.ch/jira/browse/ROOT-10293).

At this point review also [this](https://github.com/AIDASoft/DD4hep/commit/f225a01955d94f1679178174b78e9e51005254e1) implementation.

### LINKED ISSUE #607: DDRec.py broken
- OS version: SLC6
 - Compiler version: GCC 8.2
 - DD4hep version: v01-11, iLCSoft version v02-01-pre from Desy AFS (testing phase)
 - Goal: Running the calibration software LCCalibration
 - Python version: 2.7.16

## Issue

I can't import `DDRec.py` correctly anymore.

Minimal broken example to reproduce the issue (on a SL6 machine) :

```shell
source /afs/desy.de/project/ilcsoft/sw/x86_64_gcc82_sl6/v02-01-pre/init_ilcsoft.sh
python
```
then in python:

```python
import ROOT
from ROOT import gROOT
gROOT.SetBatch(1)

try:
    import DD4hep                   
except ImportError:
    import dd4hep as DD4hep

import DDRec
```

I get the following error on `import DDRec`:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/afs/desy.de/project/ilcsoft/sw/x86_64_gcc82_sl6/v02-01-pre/root/6.18.04/lib/ROOT.py", line 533, in _importhook
    return _orig_ihook( name, *args, **kwds )
  File "/afs/desy.de/project/ilcsoft/sw/x86_64_gcc82_sl6/v02-01-pre/DD4hep/v01-11/lib/python2.7/site-packages/DDRec.py", line 101, in <module>
    std_list_ISurface = core.std_list('ISurface*')
  File "/afs/desy.de/project/ilcsoft/sw/x86_64_gcc82_sl6/v02-01-pre/root/6.18.04/lib/cppyy.py", line 130, in __call__
    result = _backend.MakeRootTemplateClass( *newargs )
SyntaxError: could not get __cppname__ from provided template argument. Is it a str, class, type or int?
```

It was working with iLCSoft v02-00-02 (production version) with DD4hep v01-07-02, as we ran our energy calibration with this version.

### PR DESCRIPTION
**Pull request by petricm**
Resolves #607 and resolves #563
BEGINRELEASENOTES
- Add missing conversion from `unicode` to `str` for python 2 case in `DDrec.py`
- Add test to check dd4hep python module imports
- Fixes for tests from examples:
   - typo in tessellated example
   - make gdml read test dependent on the gdml write test
- Add check to test if the version of python used to build ROOT is the same to the version detected by CMake to build DD4hep (works only from ROOT 6.20)

ENDRELEASENOTES

### CODE REVIEW

**Comment by andresailer** [comment]
Can you add a test, it would basically only have to do importing of DD*.py, I think

**Comment by petricm** [comment]
> Can you add a test, it would basically only have to do importing of DD*.py, I think

Done, however this uncovered #623

**Comment by andresailer** [comment]
Great, thanks!
`s/however/and/` , this is a good thing :)

File: examples/ClientTests/compact/Check_Shape_Tesselated.xml [hunk 1]
```diff
@@ -195,7 +195,7 @@
         <position x="0"  y="0" z="0"/>
         <rotation x="0"   y="0" z="0"/>
       </check>
-      <test  type="DD4hep_Mesh_Verifier" ref="${DD4hepExamplesINSTALL}/ClientTests/ref/Ref_Tessellated.txt" create="CheckShape_create"/>
+      <test  type="DD4hep_Mesh_Verifier" ref="${DD4hepExamplesINSTALL}/examples/ClientTests/ref/Ref_Tessellated.txt" create="CheckShape_create"/>
```
Comment (andresailer) [inline examples/ClientTests/compact/Check_Shape_Tesselated.xml]:
```
examples/ClientTests/ref/Ref_Tesselated.txt
```
Too few L's in the filename (or too many in the XML)

**Pull request merged**